themes/mongodb/src/js/componentTabs.js

```diff
@@ -13,21 +13,60 @@ class TabsSingleton {
     constructor(key) {
         this.key = key;
         this.tabStrip = document.querySelector('.tab-strip--singleton');
+
+        // Only tab sets will have a type, init and try to retrieve
+        this.type = null;
+        if (this.tabStrip !== null) {
+            this.type = this.tabStrip.getAttribute('data-tab-preference');
+        }
     }
 
-    get languagePref() {
-        return window.localStorage.getItem(this.key);
+    /**
+     * Return the tabPref object containing preferences for tab sets
+     * and page specific prefs. Returns an empty object if it doesn't
+     * exist.
+     * @returns {object} Tab preference object.
+     */
+    get tabPref() {
+        let pref = JSON.parse(window.localStorage.getItem(this.key));
+
+        if (!pref) {
+            pref = {};
+        }
+
+        return pref;
     }
 
-    set languagePref(value) {
-        window.localStorage.setItem(this.key, value);
+    /**
+     * Sets the tabPref object depending on whether the tab belongs
+     * to set (e.g., "drivers") or if it's a one-off page.
+     * @param {object} value The "tabId" and optional "type" (tab set)
+     */
+    set tabPref(value) {
+        const tabPref = this.tabPref;
```

Comment:
I'd not make this local variable

Reply:
Discussed offline, calling the getter method above.